File: ProblemTwo/ProblemTwo.lua
--[[
	Project Euler - Problem Two.
	Author: Ronald Zielaznicki
	Problem: By considering the terms in the Fibonacci sequence whose 
	values do not exceed four million, find the sum of the even-valued 
	terms.
--]]

function sumOfEvenFibs(limit)
	sumReturn = 0;
	j = 0
	i = 1
	while (i < limit)  do
		if(i % 2 == 0) then
			sumReturn = sumReturn + i;
		end
		i = i + j;
		j = i - j;
	end
	
	return sumReturn;
end

print("Answer: ".. sumOfEvenFibs(4000000));

File: ProblemTwo/ProblemTwo.js
/*
	Project Euler - Problem Two.
	Author: Ronald Zielaznicki
	Problem: By considering the terms in the Fibonacci sequence whose 
	values do not exceed four million, find the sum of the even-valued 
	terms.
*/

function sumOfEvenFibs(limit){
	var sumReturn = 0;
	
	for(var i = 1, j = 0; i < limit; i +=j, j = i - j){
		if(i % 2 == 0)
			sumReturn += i;
	}
	
	return sumReturn;
}

console.log("Answer: " + sumOfEvenFibs(4000000));